packages/vsx-registry/src/browser/vsx-extensions-model.ts

```diff
@@ -225,49 +225,59 @@ export class VSXExtensionsModel {
-                const id = data.namespace.toLowerCase() + '.' + data.name.toLowerCase();
-                const allVersions = filter.getLatestCompatibleVersion(data);
-                if (!allVersions) {
-                    continue;
+            try {
+                const result = await client.search(param);
+
+                if (token.isCancellationRequested) {
+                    return;
                 }
-                if (this.preferences.get('extensions.onlyShowVerifiedExtensions')) {
-                    this.fetchVerifiedStatus(id, client, allVersions).then(verified => {
-                        this.doChange(() => {
-                            this.addExtensions(data, id, allVersions, !!verified);
-                            return Promise.resolve();
+                for (const data of result.extensions) {
+                    const id = data.namespace.toLowerCase() + '.' + data.name.toLowerCase();
+                    const allVersions = filter.getLatestCompatibleVersion(data);
+                    if (!allVersions) {
+                        continue;
+                    }
+                    if (this.preferences.get('extensions.onlyShowVerifiedExtensions')) {
+                        this.fetchVerifiedStatus(id, client, allVersions).then(verified => {
+                            this.doChange(() => {
+                                this.addExtensions(data, id, allVersions, !!verified);
+                                return Promise.resolve();
+                            });
                         });
-                    });
-                } else {
-                    this.addExtensions(data, id, allVersions);
-                    this.fetchVerifiedStatus(id, client, allVersions).then(verified => {
-                        this.doChange(() => {
-                            let extension = this.getExtension(id);
-                            extension = this.setExtension(id);
-                            extension.update(Object.assign({
-                                verified: verified
-                            }));
-                            return Promise.resolve();
+                    } else {
+                        this.addExtensions(data, id, allVersions);
+                        this.fetchVerifiedStatus(id, client, allVersions).then(verified => {
+                            this.doChange(() => {
+                                let extension = this.getExtension(id);
+                                extension = this.setExtension(id);
+                                extension.update(Object.assign({
+                                    verified: verified
+                                }));
+                                return Promise.resolve();
+                            });
                         });
-                    });
+                    }
                 }
+            } catch (error) {
+                this._searchError = error;
```

Comment:
Suggestion: We should stringify this error somehow, as `_searchError` is defined as a string. (while `error` is `any` and likely of class `Error`)

Reply:
The equivalent of what we had before would be `err?.message || String(err),`.